backend/src/main/java/com/nikogrid/backend/services/JwtTokenService.java

```diff
@@ -0,0 +1,56 @@
+package com.nikogrid.backend.services;
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.algorithms.Algorithm;
+import com.auth0.jwt.exceptions.JWTCreationException;
+import com.auth0.jwt.exceptions.JWTVerificationException;
+import com.nikogrid.backend.entities.UserDetailsImpl;
+import com.nikogrid.backend.exceptions.TokenGenerationException;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+
+@Service
+public class JwtTokenService {
+
+    @Value("${api.security.token.secret}")
+    private String secret;
+
+    public String generateToken(UserDetailsImpl user) {
+        try{
+            Algorithm algorithm = Algorithm.HMAC256(secret);
+            return JWT.create()
+                    .withIssuer("auth-api")
+                    .withIssuedAt(creationDate())
+                    .withExpiresAt(generateExpirationTime())
+                    .withSubject(user.getUsername())
+                    .sign(algorithm);
+        } catch (JWTCreationException e) {
+            throw new TokenGenerationException("Error while generating token", e);
+        }
+    }
+
+    public String validateToken(String token) {
+        try {
+            Algorithm algorithm = Algorithm.HMAC256(secret);
```

Comment:
You seem to duplicate this. Transform this into a private static final variable and use in both functions.